test/orange/g4org/SolidConverter.test.cc

```diff
@@ -907,26 +906,41 @@ TEST_F(SolidConverterTest, polyhedra)
+            celeritas::DebugError);
```

Comment:
DebugError is for programming errors (assertion failures, precondition violations, bad assumptions) whereas RuntimeError is for things like malformed user input and system status that we can't know about until runtime
```suggestion
            celeritas::RuntimeError);
```

Reply:
This hits a programming error that I am yet to debug, hence the `DebugError`

Reply:
I thought it was a shortcoming that resulted in a `CELER_VALIDATE` error throwing? Since DebugError throws should (with very very few exceptions) be disabled in release builds, any GoogleTest that checks for a DebugError needs to have an `if (CELERITAS_DEBUG)` wrapping it.

Also the reason I said RuntimeError was that I was seeing one in the test output, so there must be a secondary failure as well:
```
/home/runner/work/celeritas/celeritas/test/orange/g4org/SolidConverter.test.cc:942
Expected: this->build_and_test( G4Polyhedra("HGCalHEAbsorber110x7f1fff5a7880", 0 * deg, 360 * deg, 36, std::size(z2), z2, rmin2, rmax2), R"json({"_type":"all","daughters":[{"_type":"stackedextrudedpolygon","polygon":[[1.0,0.0],[0.984807753012208,0.17364817766693033],[0.9396926207859084,0.3420201433256687],[0.8660254037844387,0.5],[0.766044443118978,0.6427876096865393],[0.6427876096865393,0.766044443118978],[0.5,0.8660254037844386],[0.3420201433256689,0.9396926207859083],[0.17364817766693044,0.984807753012208],[0.0,1.0],[-0.17364817766693044,0.984807753012208],[-0.34202014332566855,0.9396926207859084],[-0.4999999999999999,0.8660254037844387],[-0.6427876096865393,0.766044443118978],[-0.7660444431189778,0.6427876096865396],[-0.8660254037844385,0.5000000000000001],[-0.9396926207859083,0.3420201433256689],[-0.984807753012208,0.17364817766693044],[-1.0,0.0],[-0.984807753012208,-0.17364817766693044],[-0.9396926207859083,-0.3420201433256689],[-0.8660254037844389,-0.49999999999999956],[-0.7660444431189783,-0.642787609686539],[-0.6427876096865396,-0.7660444431189778],[-0.5000000000000001,-0.8660254037844385],[-0.3420201433256689,-0.9396926207859083],[-0.17364817766693044,-0.984807753012208],[0.0,-1.0],[0.17364817766692975,-0.9848077530122081],[0.3420201433256682,-0.9396926207859085],[0.49999999999999956,-0.8660254037844389],[0.642787609686539,-0.7660444431189783],[0.7660444431189778,-0.6427876096865396],[0.8660254037844385,-0.5000000000000001],[0.9396926207859083,-0.3420201433256689],[0.984807753012208,-0.17364817766693044]],"polyline":[[0.0,0.0,-2.075],[0.0,0.0,2.0740000000000203],[0.0,0.0,2.07400000000012],[0.0,0.0,2.07499999999995]],"scaling":[199.16630529221047,204.42053956048494,204.42053956048494,204.42180594618483]},{"_type":"negated","daughter":{"_type":"stackedextrudedpolygon","polygon":[[1.0,0.0],[0.984807753012208,0.17364817766693033],[0.9396926207859084,0.3420201433256687],[0.8660254037844387,0.5],[0.766044443118978,0.6427876096865393],[0.6427876096865393,0.766044443118978],[0.5,0.8660254037844386],[0.3420201433256689,0.9396926207859083],[0.17364817766693044,0.984807753012208],[0.0,1.0],[-0.17364817766693044,0.984807753012208],[-0.34202014332566855,0.9396926207859084],[-0.4999999999999999,0.8660254037844387],[-0.6427876096865393,0.766044443118978],[-0.7660444431189778,0.6427876096865396],[-0.8660254037844385,0.5000000000000001],[-0.9396926207859083,0.3420201433256689],[-0.984807753012208,0.17364817766693044],[-1.0,0.0],[-0.984807753012208,-0.17364817766693044],[-0.9396926207859083,-0.3420201433256689],[-0.8660254037844389,-0.49999999999999956],[-0.7660444431189783,-0.642787609686539],[-0.6427876096865396,-0.7660444431189778],[-0.5000000000000001,-0.8660254037844385],[-0.3420201433256689,-0.9396926207859083],[-0.17364817766693044,-0.984807753012208],[0.0,-1.0],[0.17364817766692975,-0.9848077530122081],[0.3420201433256682,-0.9396926207859085],[0.49999999999999956,-0.8660254037844389],[0.642787609686539,-0.7660444431189783],[0.7660444431189778,-0.6427876096865396],[0.8660254037844385,-0.5000000000000001],[0.9396926207859083,-0.3420201433256689],[0.984807753012208,-0.17364817766693044]],"polyline":[[0.0,0.0,-2.075],[0.0,0.0,2.0740000000000203],[0.0,0.0,2.07400000000012],[0.0,0.0,2.07499999999995]],"scaling":[34.99315953676109,34.99315953676109,42.019898399564525,42.019898399564525]},"label":""}],"label":"HGCalHEAbsorber110x7f1fff5a7880"})json") throws an exception of type celeritas::DebugError.
  Actual: it throws celeritas::RuntimeError with description "runtime error: z coordinates must be strictly increasing
/home/runner/work/celeritas/celeritas/src/orange/orangeinp/StackedExtrudedPolygon.cc:103: 'std::adjacent_find(polyline_.begin(), polyline_.end(), [](auto const& a, auto const& b) { return a[Z] >= b[Z]; }) == polyline_.end()' failed".
```

Reply:
Both CMS tests throw a `RuntimeError` for having zero-height `z` segments. The first one goes away if you bump, which what I was originally doing, but I took it out because 1) bumping is hack anyway, 2) I converted to `EXPECT_THROW`. The second CMS test hits a `CELER_ASSERT` even with bumping. I will wrap the second with `if (CELERITAS_DEBUG)`.